### ISSUE TITLE
sendline error in Spyder3 IDE

### ISSUE THREAD

**Issue by raczben** [raczben/wexpect#1]
**Issue:** Wexpect's sendline method fails calling from Spyder3 IDE. 

**Reproduce:** Run `cmd.py` example code from `exampes` folder from Spyder.

```
runfile('D:/bt/wexpect/examples/cmd.py', wdir='D:/bt/wexpect/examples')
Microsoft Windows [Version 10.0.17763.437]
(c) 2018 Microsoft Corporation. All rights reserved.

D:\bt\wexpect\examples>[PyINPUT_RECORD(EventType=1) (KEY_EVENT), PyINPUT_RECORD(EventType=1) (KEY_EVENT)]
{'Size': PyCOORDType(X=80,Y=16000), 'CursorPosition': PyCOORDType(X=23,Y=3), 'Attributes': 7, 'Window': PySMALL_RECTType(Left=0,Top=5,Right=79,Bottom=29), 'MaximumWindowSize': PyCOORDType(X=80,Y=64)}
Traceback (most recent call last):

  File "<ipython-input-1-5e47c42ff79e>", line 1, in <module>
    runfile('D:/bt/wexpect/examples/cmd.py', wdir='D:/bt/wexpect/examples')

  File "C:\Users\usr\AppData\Local\Programs\Python\Python37\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 704, in runfile
    execfile(filename, namespace)

  File "C:\Users\usr\AppData\Local\Programs\Python\Python37\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 108, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "D:/bt/wexpect/examples/cmd.py", line 30, in <module>
    p.sendline('ls')

  File "C:\Users\usr\AppData\Local\Programs\Python\Python37\lib\site-packages\wexpect.py", line 997, in sendline
    n = self.send(s)

  File "C:\Users\usr\AppData\Local\Programs\Python\Python37\lib\site-packages\wexpect.py", line 1824, in send
    c = self.wtty.write(s)

  File "C:\Users\usr\AppData\Local\Programs\Python\Python37\lib\site-packages\wexpect.py", line 2140, in write
    wrote = self.__consin.WriteConsoleInput(records)

error: (6, 'WriteConsoleInput', 'The handle is invalid.')
```

**Comment by sbelizaire** [comment]
I am getting the same error. How did you solve this?

**Comment by raczben** [comment]
Sorry, No, I didn't solve this. I have no idea yet.

**Comment by raczben** [comment]
The same issue happens using pyCharm.

**Comment by raczben** [comment]
https://github.com/raczben/wexpect/projects/1

**Comment by raczben** [comment]
149ef50 is an initial solution. Basic use-cases (spawn, send, expect) works in Spyder IDE, more feature will be added soon.

**Comment by raczben** [comment]
[v3.2.0](https://github.com/raczben/wexpect/releases/tag/v3.2.0) solve this if `WEXPECT_SPAWN_CLASS` environment variable is set to `SpawnPipe` or `SpawnSocket`:

```
export WEXPECT_SPAWN_CLASS=SpawnSocket
```

**Comment by raczben** [comment]
Solved with [v4.0.0](https://github.com/raczben/wexpect/releases/tag/v4.0.0).
